kobo/worker/taskmanager.py

```diff
@@ -87,7 +88,30 @@ class TaskContainer(PluginContainer):
     pass
 
 
-class TaskManager(kobo.log.LoggingBase):
+@six.add_metaclass(abc.ABCMeta)
+class TaskManagerBase(kobo.log.LoggingBase):
+    """Abstract class for a task manager defining which methods must be implemented."""
+    @abc.abstractmethod
+    def update_worker_info(self):
```

Comment:
It would be friendly to put some doc strings onto these explaining the behavior that the implementer should aim for...  it's not obvious from the names alone, and I assume that you had to figure that out already per method when subclassing.